### PR TITLE
[KEYCLOAK-10675] - Deleting an Identity Provider doesn't remove the associated IdP Mapper for that user

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: server-spi/src/main/java/org/keycloak/models/UserProvider.java [hunk 1]
```diff
@@ -40,6 +40,7 @@
 
     void addFederatedIdentity(RealmModel realm, UserModel user, FederatedIdentityModel socialLink);
     boolean removeFederatedIdentity(RealmModel realm, UserModel user, String socialProvider);
+    void removeFederatedIdentity(RealmModel realm, String socialProvider);
```
Comment (mposolda) [inline server-spi/src/main/java/org/keycloak/models/UserProvider.java]: Just minor concern, but from the name of this method, it is not obvious what is is doing... Is it possible to change the method signature to "preRemove(RealmModel, IdentityProviderModel)" to be compatible with all the other similar methods on the UserProvider interface (methods which intercepts removing on some realm object) ? WDYT?

**Review by mposolda — CHANGES_REQUESTED** [review]
@pedroigor LGTM, just one minor concern regarding the name of the method.

File: server-spi/src/main/java/org/keycloak/models/UserProvider.java (hunk 1 above)
Comment (pedroigor) [reply server-spi/src/main/java/org/keycloak/models/UserProvider.java]: Sure

Comment (pedroigor) [reply server-spi/src/main/java/org/keycloak/models/UserProvider.java]: @mposolda I've also added more assertion to tests.

**Review by mposolda — APPROVED** [review]
@pedroigor Thanks!

**Pull request merged**